### ISSUE TITLE
vim9class: null_object - constant type 17 not supported

### ISSUE THREAD

**Issue by errael** [vim/vim#12043]
### Steps to reproduce

(It may be that null_object is only partially implemented, and so this is not unexpected)

When there's clarification, I can try expected behavior

```
vim9script

# OK
var nullo = null_object
def F(): any
    return nullo
enddef

# Maybe OK: output: type(F()) 13
echo "type(F())" type(F())

# Maybe OK: output: typename(F()) object<Unknown>
echo "typename(F())" typename(F())

# OK
var o1: any
o1 = nullo

# OK
echo o1 == null_object ? "o1 == null_object" : "o1 != null_object"

# FAIL: expect o1 == null
echo o1 == null ? "o1 == null" : "o1 != null"

# FAIL compilation error
### This works as of patch 9.0.1357
def G()
    var x = null_object
enddef

class C
endclass

# FAIL
var o2: C
o2 = null_object
```

### Expected behaviour

The expected results are indicated in the steps to reproduce.

### Version of Vim

9.0.1333

### Environment

ubuntu

**Comment by errael** [comment]
This issue has several examples of using null_object. 3 of them failed when the issue was filed. One of the failures now works. In particular
```
        def G()
            var x = null_object
        enddef
```
Used to fail compilation. It was fixed by https://github.com/vim/vim/commit/c4e1b86cb0d88fa5ec1141d3c600e026dcc1bc21

**Comment by errael** [comment]
Some additional output and comments for the last failure
```
class C
endclass

# FAIL
var o2: C
echo "o2:" o2
echo "o2:" typename(o2)
o2 = null_object
```
there's the output
```
    o2: object of [unknown]
    o2: object<Unknown>
    Error detected while processing command line..script /.../12043-b.vim:
    line    9:
    E1012: Type mismatch; expected object<C> but got object<Unknown>
```
It seems like a bug, otherwise there would be no way to null out an object, like for garbage collection. As a workaround, I guess one could craft an empty object of the correct type and use it as a null analog.

**Comment by errael** [comment]
And there's
```
$ srcgrep null_class
./src/eval.c:	case 10: if (STRNCMP(s, "null_class", 10) == 0)
./src/testdir/test_vim9_class.vim:def Test_using_null_class()
./src/testdir/test_vim9_class.vim:      @_ = null_class.member
```

**Comment by yegappan** [comment]
Please feel free to work on this.  I think PR #12867 has some changes that will be needed here. I started working on adding the test_null_object() and test_null_class() functions. These will be useful in writing tests.

**Comment by errael** [comment]
> started working on adding the test_null_object() and test_null_class() functions

I don't see any reference to `test_null_*` in PR #12867 but I do see quite a bit about class/object which I guess are steps on the way. Looks like that PR is very close, it will probably be in before I'm ready to go...

**Comment by yegappan** [comment]
Now that PR #12890 is merged, can this issue be closed?

**Comment by errael** [comment]
Closed by #12890.
